getting-started/quickstart-guides/php.md

```diff
@@ -42,425 +42,1483 @@ It will prompt a few options. Select defaults by pressing enter.
-* `publicKey` and `privateKey` parameters are also required as these would be used for all ImageKit API, server-side upload, and generating token for client-side file upload. You can get these parameters from the developer section in your ImageKit dashboard - [https://imagekit.io/dashboard#developers](https://imagekit.io/dashboard#developers).
-
-## Generating url for rendering images
-
-#### URL for image with relative path&#x20;
-
+#### URL Generation
 ```php
-$imageURL = $imageKit->url(['path' => '/default-image.jpg']);
-
-echo('Url : ' . $imageURL); 
-// https://ik.imagekit.io/demo/default-image.jpg
+// For URL Generation
+$imageURL = $imageKit->url(
+    [
+        'path' => '/default-image.jpg',
+    ]
+);
+echo $imageURL;
 ```
 
-![](../../.gitbook/assets/default-image.jpeg)
-
-#### URL for image with relative path and custom URL-endpoint
-
+#### File Upload
 ```php
-$imageURL = $imageKit->url([
-      'urlEndpoint' => 'https://ik.imagekit.io/test',
-      'path' => '/default-image.jpg',
+// For File Upload
+$uploadFile = $imageKit->upload([
+    'file' => 'file-url',
+    'fileName' => 'new-file'
 ]);
+```
 
-echo('Url : ' . $imageURL); 
-// https://ik.imagekit.io/test/default-image.jpg
+#### Response Structure
+```json
+{
+    "error": {},
+    "result": {},
+    "responseMetadata": {
+        "headers": {},
+        "raw": {},
+        "statusCode": {statusCode}
+    }
+}
 ```
 
-#### &#x20;URL for image with absolute url&#x20;
+## Generating url for rendering images
+ImageKit provides inbuild media storage and integration with external origins. Refer to the [Documentation](https://docs.imagekit.io/integration/url-endpoints) to learn more about URL endpoints and external [Image Origins](https://docs.imagekit.io/integration/configure-origin) supported by ImageKit.  
+
+### Using Image path and image hostname or endpoint 
+  
+This method allows you to create a URL using the image's path and the ImageKit URL endpoint (urlEndpoint) you want to use to access the image.   
+  
+#### Example
+```php  
+$imageURL = $imageKit->url(
+    [
+        'path' => '/default-image.jpg', 
+        'transformation' => [
+            [
+                'height' => '300', 
+                'width' => '400'
+            ]
+        ]
+    ]
+);
+```  
 
-If you have an absolute image path coming from the backend API e.g. `https://www.custom-domain.com/default-image.jpg` then you can use `src` prop to load the image.
+#### Response
 
-```php
+```  
+https://ik.imagekit.io/demo/tr:h-300,w-400/default-image.jpg 
+```  
+
+
+### Using full image URL
+This method allows you to add transformation parameters to an absolute ImageKit powered URL. This method should be used if you have the absolute URL stored in your database.
+
+#### Example
+```php  
 $imageURL = $imageKit->url([
-    'src' => 'https://example.com/default-image.jpg'
+    'src' => 'https://ik.imagekit.io/your_imagekit_id/endpoint/default-image.jpg',
```

Comment:
`https://example.com/default-image.jpg` was more appropriate to differentiate it from other examples. Let it be that.